File: QytetetRuby/lib/sorpresa.rb
# encoding: utf-8

module ModeloQytetet
  
  # Esta clase representa una "carta sorpresa"
  # de Qytetet
  
  class Sorpresa
    attr_reader:texto
    attr_reader:valor
    attr_reader:tipo

    # Constructor que crea una carta sorpesa a partir de 3 argumentos
    
    def initialize (texto, valor, tipo)
      @texto = texto
      @valor = valor
      @tipo = tipo
    end
    
    # Método que imprime los atributos de una carta sorpresa

    def to_s()
      "Texto; #{@texto} \n Valor: #{@valor} \n Tipo: #{@tipo}"
    end 

  end
end

File: QytetetJava/src/modeloqytetet/Sorpresa.java
package modeloqytetet;

    /* 
     Esta clase cada una de
     las cartas sorpresa de
     qytetet
    */

public class Sorpresa {
    
 // Atributos privados
    
        // Mensaje de la carta
    private String texto;
        // Tipo de sopresa
    private TipoSorpresa tipo;
        // Valor (puede ser 0)
    private int valor;
    
        
        // Constructor 
    
    Sorpresa (String texto, int valor, TipoSorpresa tipo){
        this.texto = texto;
        this.tipo = tipo;
        this.valor = valor;
    }
    
        // Consultores
    
    String getTexto (){return (texto);} 
    
    TipoSorpresa getTipo (){return (tipo);}
    
    int getValor (){return (valor);}
    
        // Sorpresa toString()
    
    @Override
    public String toString (){
        return texto + 
                "\nValor: " + Integer.toString(valor) + 
                "\nTipo=" + tipo;
    }
}
